quizmaster-client-app/src/components/HomePage/HomePage.tsx

```diff
@@ -85,13 +85,18 @@ function HomePage() {
 
   const onJoinQuizSubmit = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-    Axios.get(`/api/quizzes/${quizCode}/name`).then((res) => {
-      if (!res.data) {
+    Axios.get(`/api/quizzes/${quizCode}/name`)
+      .then((res) => {
+        if (!res.data) {
+          setOpen(true);
+        } else {
+          history.push(`/quiz/${quizCode}/${res.data.toUrlFormat()}`);
+        }
+      })
+      .catch((err) => {
+        console.log("error: ", err);
```

Comment:
I don't think we should commit logs to the console. We could also check for a 404 error using `err.response.status` e.g.
```
if (err.response.status === 404) {
      throw new Error(`${err.config.url} not found`);
}
```

Reply:
Is error 404 the code returned when the quiz code is not valid, or is error 404 a different error?

Reply:
Just checked and it is 404 ,   but what happens if its an error other than 404?  In this case the error will be "caught" and then not handled,  perhaps we need an ELSE here to throw the exception if its something other than 404

Reply:
Calum's mentioned that there might be other errors which would probably be handled globally (i.e. a server error even if the quiz code is correct etc.) and can be sorted latet